### PR TITLE
Fix members page infinite loop

### PR DESCRIPTION
**Pull request by rbennettcw**
## Link to Issue
Closes: #7155

## Description of Changes
- Changes to the community members page
  - ~~Added throttling to limit to 1 page fetch per second~~
  - Disabled the `CWTable` internal sorting– the table's internal sorting didn't match the API call's sorting, which resulted in infinite loop and unpredictable behavior
  - Fixed the search bar behavior– it was not passing the query to the API call previously

## "How We Fixed It"
- The `CWTable` has in-memory sorting, which conflicted with the `orderBy` param set for the `useSearchProfilesQuery` API call.
  - When new items were added to the list, the items were rendered in an unpredictable way, with items always rendered "above the fold", resulting infinite loop of fetching pages.
  - Solution was to disable sorting for the `CWTable` and rely on the order provided by the API response. Use a single source of truth for order.

## Test Plan
- CA test
  - Go to http://localhost:8080/ethereum/members?tab=all-members
  - Continuously scroll down for ~10 seconds– confirm that pages are fetched without rapid fire API calls
  - Use the search bar to search for profile. Confirm it works properly.

## Deployment Plan
N/A

## Other Considerations
- This is a workout that comes at the cost of disabled sorting on the members page. It's set to always sort by last activity.
- A proper fix would require making the `CWTable` a controlled component– because currently it contains sorting state internally. That means we cannot have a single source of truth for order column without refactoring the `CWTable` component, which is used in 5 different places.
  - Will create FF ticket for this.

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/pages/CommunityGroupsAndMembers/Members/MembersSection/MembersSection.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/CommunityGroupsAndMembers/Members/MembersSection/MembersSection.tsx:1]: I tried `sortable: true` with the other changes in this PR and wasn't able to repro 
> Disabled the CWTable internal sorting– the table's internal sorting didn't match the API call's sorting, which resulted in infinite loop and unpredictable behavior

can you confirm if it's still reproducible for you @rbennettcw?

https://github.com/hicommonwealth/commonwealth/assets/51641047/11979330-47ad-4674-a732-e31bd135b2ae

Comment (rbennettcw) [reply packages/commonwealth/client/scripts/views/pages/CommunityGroupsAndMembers/Members/MembersSection/MembersSection.tsx:1]: @mzparacha Your video actually *did* reproduce part of the problem 🙂 – notice how after a few seconds of scrolling down, there were seemingly fewer and fewer items added to the bottom, and the list changed in a random way when content was loaded in.

That's not normal– it's supposed to keep adding 10+ items at the bottom until it runs out. The existing items in the list are not supposed to change order when you scroll.

---

What you're seeing is that the throttle helped to limit the number of requests– which partially solved the issue– but if sorting is enabled, it still screws up the way that new items are added to the list.

---

Bottom line is that the CWTable should not sort items in memory. It should update the `orderBy` and `orderDirection` state of the API call, and let the API handle sorting.

Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/CommunityGroupsAndMembers/Members/MembersSection/MembersSection.tsx:1]: > notice how after a few seconds of scrolling down, there were seemingly fewer and fewer items added to the bottom

i think it's because of the ascending sort on the column, descending sort does the opposite (adds more and more items to bottom as they are fetched)

Comment (rbennettcw) [reply packages/commonwealth/client/scripts/views/pages/CommunityGroupsAndMembers/Members/MembersSection/MembersSection.tsx:1]: Sure, maybe. But do we want to have sorting that's broken/working depending on how you sort? I think it's best to disable sorting on the CWTable until we change it into a controlled component to properly handle state.

Backend pagination + client-side sorting doesn't work. Since the API controls pagination, it should also control sorting.

Comment (rbennettcw) [reply packages/commonwealth/client/scripts/views/pages/CommunityGroupsAndMembers/Members/MembersSection/MembersSection.tsx:1]: FYI: The API call sorts by Last Active DESC, which the table doesn't even support currently. So regardless of how the table is sorted, it won't properly add items to the list, even if it *looks* like it does.

**Review by lzach83 — APPROVED** [review]
Approving, have to fix the CI errors. Got caught with the eslint-diff

**Comment by rbennettcw** [comment]
Removed throttle since it wasn't needed, and not sure if it was even working in the first place. The sorting fix made throttling obsolete anyway.

**Review by jnaviask — APPROVED** [review]
Fine as a quick fix so long as we can get the controlled component up to reenable sorting.
